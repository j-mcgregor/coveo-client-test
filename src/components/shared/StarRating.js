import PropTypes from 'prop-types';
import React from 'react';
import StarRatings from 'react-star-ratings';

const StarRating = ({ percentScore }) => {
  const float = Math.floor(percentScore) / 20;

  return (
    <div>
      <StarRatings
        rating={float}
        starRatedColor="#782c42"
        numberOfStars={5}
        starDimension="12px"
        starSpacing="0.5px"
      />
    </div>
  );
};

StarRating.propTypes = {
  percentScore: PropTypes.number
};

StarRating.defaultProps = {
  percentScore: null
};

export default StarRating;
