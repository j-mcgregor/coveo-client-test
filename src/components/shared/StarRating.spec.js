import React from 'react';
import { shallow } from 'enzyme';
import StarRating from './StarRating';

describe('StarRating', () => {
  it('should render the StarRating', () => {
    const props = {
      percentScore: 84
    };
    const wrapper = shallow(<StarRating {...props} />);

    expect(wrapper).toMatchSnapshot();
  });
});
