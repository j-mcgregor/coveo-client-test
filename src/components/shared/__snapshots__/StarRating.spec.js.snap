// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StarRating should render the StarRating 1`] = `ShallowWrapper {}`;
